File: app/src/main/java/info/nightscout/androidaps/plugins/NSClientInternal/data/NSAlarm.java
package info.nightscout.androidaps.plugins.NSClientInternal.data;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by mike on 11.06.2017.
 */

public class NSAlarm {
    JSONObject data;

    public NSAlarm(JSONObject data) {
        this.data = data;
    }

    public int getLevel() {
        int retval = 0;
        if (data.has("level")) {
            try {
                retval = data.getInt("level");
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
         return retval;
    }

    public String getGroup() {
        String retval = "N/A";
        if (data.has("group")) {
            try {
                retval = data.getString("group");
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
         return retval;
    }

    public String getTile() {
        String retval = "N/A";
        if (data.has("title")) {
            try {
                retval = data.getString("title");
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
         return retval;
    }

    public String getMessage() {
        String retval = "N/A";
        if (data.has("message")) {
            try {
                retval = data.getString("message");
            } catch (JSONException e) {
                e.printStackTrace();
            }
        }
         return retval;
    }
}

File: plugins/sync/src/main/java/info/nightscout/plugins/sync/nsShared/NSAlarm.kt
package info.nightscout.plugins.sync.nsShared

import info.nightscout.core.utils.JsonHelper
import info.nightscout.interfaces.nsclient.NSAlarm
import org.json.JSONObject

class NSAlarmObject(private var data: JSONObject) : NSAlarm {
    /*
    {
    "level":2,
    "title":"Urgent HIGH",
    "message":"BG Now: 5.2 -0.1 → mmol\/L\nRaw BG: 5 mmol\/L Čistý\nBG 15m: 5 mmol\/L\nIOB: 0.00U\nCOB: 0g",
    "eventName":"high",
    "plugin":{"name":"simplealarms","label":"Simple Alarms","pluginType":"notification","enabled":true},
    "pushoverSound":"persistent",
    "debug":{"lastSGV":5.2,"thresholds":{"bgHigh":80,"bgTargetTop":75,"bgTargetBottom":72,"bgLow":70}},
    "group":"default",
    "key":"simplealarms_2"
    }
     */

    override fun level(): Int =
        JsonHelper.safeGetInt(data, "level", 0)

    override fun group(): String =
        JsonHelper.safeGetString(data, "group", "N/A")

    override fun title(): String =
        JsonHelper.safeGetString(data, "title", "N/A")

    override fun message(): String =
        JsonHelper.safeGetString(data, "message", "N/A")

    override fun low(): Boolean =
        JsonHelper.safeGetString(data, "eventName", "") == "low"

    override fun high(): Boolean =
        JsonHelper.safeGetString(data, "eventName", "") == "high"

    override fun timeago(): Boolean =
        JsonHelper.safeGetString(data, "eventName", "") == "timeago"
}